### ISSUE TITLE
tf.contrib.image.transform crashes under Windows when CUDA is enabled

### ISSUE THREAD

**Issue by hrschubert** [tensorflow/tensorflow#17485]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
Yes. A minimal example reproducing the bug is provided below.
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Windows 10 (x64)
- **TensorFlow installed from (source or binary)**: 
Binary installed using `pip install tensorflow-gpu`
- **TensorFlow version (use command below)**: 
b'unknown' 1.6.0 (also tested on b'unknown' 1.4.0)
- **Python version**: 3.6
- **Bazel version (if compiling from source)**: N/A
- **GCC/Compiler version (if compiling from source)**: N/A
- **CUDA/cuDNN version**:  
  CUDA 9.0.176 / cuDNN 7.0.5
  CUDA 8.0.61 / cuDNN 6.14.11
- **GPU model and memory**:
(device: 0, name: GeForce GTX 970M, pci bus id: 0000:01:00.0, compute capability: 5.2, memory: 3.00GiB)
(device: 0, name: Quadro M1200, pci bus id: 0000:01:00.0, compute capability: 5.0, memory: 4.00GiB)
- **Exact command to reproduce**: python example.py (see below)

### Describe the problem
The function tf.contrib.image.transform crashes when CUDA is enabled under Windows. It produces the following errors: `CUDA_ERROR_ILLEGAL_INSTRUCTION` on tensorflow 1.6 or `CUDA_ERROR_LAUNCH_FAILED` on tensorflow 1.4.
However, it functions correctly when CUDA is disabled (by setting _CUDA_VISIBLE_DEVICES_ to -1). 

I tested a variation of different parameters (such as varying the batch size, image sizes, and number of channels), but the behavior stays the same. In addition I reproduced the same error on a different machine with an older tensorflow version.

### Source code / logs
- **Code**:
```
import numpy as np
import tensorflow as tf

batch_size, image_size, channels = 1, 32, 1

data = np.zeros(
    shape=(batch_size, image_size, image_size, channels), 
    dtype=np.float32)

data_node = tf.placeholder(
    shape=(batch_size, image_size, image_size, channels),
    dtype=tf.float32)

identity = tf.constant([1, 0, 0, 0, 1, 0, 0, 0], dtype=tf.float32)
transform = tf.tile(tf.expand_dims(identity, 0), [batch_size, 1])
data_node_transformed = tf.contrib.image.transform(data_node, transform)

data_t = tf.Session().run([data_node_transformed], feed_dict={data_node: data})
```
- **Console Output**:

  Output with tensorflow 1.6 (with GeForce GTX 970M,) :

```
2018-03-06 15:42:21.578078: I C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_device.cc:1212] Found device 0 with properties:
name: GeForce GTX 970M major: 5 minor: 2 memoryClockRate(GHz): 1.038
pciBusID: 0000:01:00.0
totalMemory: 3.00GiB freeMemory: 2.48GiB
2018-03-06 15:42:21.578310: I C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_device.cc:1312] Adding visible gpu devices: 0
2018-03-06 15:42:21.890290: I C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_device.cc:993] Creating TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 2192 MB memory) -> physical GPU (device: 0, name: GeForce GTX 970M, pci bus id: 0000:01:00.0, compute capability: 5.2)
2018-03-06 15:42:22.101031: E C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\stream_executor\cuda\cuda_event.cc:49] Error polling for event status: failed to query event: CUDA_ERROR_ILLEGAL_INSTRUCTION
2018-03-06 15:42:22.101032: E C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\stream_executor\cuda\cuda_driver.cc:1110] could not synchronize on CUDA context: CUDA_ERROR_ILLEGAL_INSTRUCTION ::
2018-03-06 15:42:22.101185: F C:\tf_jenkins\workspace\rel-win\M\windows-gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_event_mgr.cc:203] Unexpected Event status: 1
```

  Output with tensorflow 1.4 (with Quadro M1200):

```
2018-03-06 15:31:17.800588: I C:\tf_jenkins\home\workspace\rel‑win\M\windows‑gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_device.cc:1030] Found device 0 with properties: 
name: Quadro M1200 major: 5 minor: 0 memoryClockRate(GHz): 1.148
pciBusID: 0000:01:00.0
totalMemory: 4.00GiB freeMemory: 3.35GiB
2018-03-06 15:31:17.803815: I C:\tf_jenkins\home\workspace\rel‑win\M\windows‑gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_device.cc:1120] Creating TensorFlow device (/device:GPU:0) ‑> (device: 0, name: Quadro M1200, pci bus id: 0000:01:00.0, compute capability: 5.0)
2018-03-06 15:31:18.731252: E C:\tf_jenkins\home\workspace\rel‑win\M\windows‑gpu\PY\36\tensorflow\stream_executor\cuda\cuda_driver.cc:1110] could not synchronize on CUDA context: CUDA_ERROR_LAUNCH_FAILED :: No stack trace available
2018-03-06 15:31:18.731258: E C:\tf_jenkins\home\workspace\rel‑win\M\windows‑gpu\PY\36\tensorflow\stream_executor\cuda\cuda_event.cc:49] Error polling for event status: failed to query event: CUDA_ERROR_LAUNCH_FAILED
2018-03-06 15:31:18.733458: F C:\tf_jenkins\home\workspace\rel‑win\M\windows‑gpu\PY\36\tensorflow\core\common_runtime\gpu\gpu_event_mgr.cc:203] Unexpected Event status: 1
```

**Comment by shivaniag** [comment]
@gunan Can you take a look?

**Comment by gunan** [comment]
I am not familiar with this code.
@mrry @zheng-xq  have you seen the above error message before?

**Comment by mrry** [comment]
Assuming the example is minimal, it looks like the failing code is (probably) in one of these files:

https://github.com/tensorflow/tensorflow/blob/60a21e25b0261369a15ca1d17505d7c3c82be967/tensorflow/contrib/image/kernels/image_ops.h
https://github.com/tensorflow/tensorflow/blob/60a21e25b0261369a15ca1d17505d7c3c82be967/tensorflow/contrib/image/kernels/image_ops.cc
https://github.com/tensorflow/tensorflow/blob/60a21e25b0261369a15ca1d17505d7c3c82be967/tensorflow/contrib/image/kernels/image_ops_gpu.cu.cc

I don't know what could be causing the problem. @ringw contributed that code, and might have some ideas.

**Comment by ringw** [comment]
Sorry, I don't have access to a Windows machine and am not familiar with the Windows development environment, let alone debugging CUDA. I don't think I can feasibly look into this issue in the near future. If anyone manages to narrow down the error on Windows, I can at least answer any questions about the code.

**Comment by mkamalov** [comment]
The same issue on GTX970 and TF 1.5.0 and 1.6.0 when using tf.contrib.image.transform method.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @shivaniag: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by eric-czech** [comment]
I had the same issue with tf.contrib.image.translate on TF 1.7.0

Windows 10
Python 3.6.4
Cuda 9
cuDNN 7

**Comment by micoene** [comment]
I have the same issue with tf.contrib.image.rotate
Windows 10
Python 3.6
Cuda 9
cuDNN 7

**Comment by geometrikal** [comment]
I have the same issue with tf.contrib.image.rotate on TF 1.7.0
Windows 10
Python 3.6
Cuda 9
cuDNN 7

It is interesting that when wrapping it with tf.data.Dataset it works:

    import numpy as np
    import tensorflow as tf
    
    batch_size, image_size, channels = 1, 32, 1
    
    data = np.zeros(
        shape=(batch_size, image_size, image_size, channels),
        dtype=np.float32)
    
    data_node = tf.placeholder(
        shape=(batch_size, image_size, image_size, channels),
        dtype=tf.float32)
    
    identity = tf.constant([1, 0, 0, 0, 1, 0, 0, 0], dtype=tf.float32)
    transform = tf.tile(tf.expand_dims(identity, 0), [batch_size, 1])
    # data_node_transformed = tf.contrib.image.transform(data_node, transform)
    
    data = tf.data.Dataset.from_tensor_slices(data)
    data = data.map(lambda x: tf.contrib.image.transform(x, transform))
    data = data.batch(1)
    iterator = data.make_one_shot_iterator()
    next_element = iterator.get_next()
    
    with tf.Session() as session:
        data_t = session.run(next_element)

**Comment by micoene** [comment]
Mine only crashes on the GPU aswell, Wrapping it with Dataset doesn't really solve the issue since it just forces it to run on the CPU : https://github.com/tensorflow/tensorflow/issues/13610

**Comment by geometrikal** [comment]
@micoene Ahhh yes, you are correct, just putting it on CPU works. It is a workaround for now...

    with tf.device('/cpu:0'):
        data_node_transformed = tf.contrib.image.transform(data_node, transform)

**Comment by CookieChief** [comment]
Same issues Geforce 1060, Windows 10, Python 3,6, Cuda 9, cuDNN 7.04

**Comment by sonho2008** [comment]
I have the same issue with tf.contrib.image.rotate on TF 1.4 on GPU (GTX 980M)
Windows 10
Python 3.5
Cuda 8.0
cuDNN 6.1

**Comment by MeteorKepler** [comment]
Same on win10, py36, cuda9.0, cudnn7, tf1.8, gtx1080

**Comment by tensorflowbutler** [comment]
Nagging Assignee @shivaniag: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by Aliyss** [comment]
Similair issue:

Windows 10
Python 3.6
Cuda 9.0
cuDNN 7.1.4

**Comment by AloshkaD** [comment]
@Aliyss I recreated a fresh conda env with cdnn, tensorflow gpu1.8, and keras gpu 2.1 installed from anaconda. Before that I downgraded my cuda from 9.2 to 9.0

**Comment by jijingtaifeng** [comment]
Similair issue:

Windows 10
Python 3.6
Cuda 9.0
cuDNN 7.1.4

I knew that I would not upgrade. I still can’t solve it now.

**Comment by Aliyss** [comment]
@jijingtaifeng I tried running it from the python console instead of Visual Studio or similair. try that. maybe it works.

**Comment by aligokalppeker** [comment]
Same issue:
Tensoflow-gpu 1.8
Windows 10
Python 3.6.2
Cuda 9.0
cuDNN 7

**Comment by BBarbosa** [comment]
Similar issue:

Windows 10
Python 3.6
Tensorflow 1.9
CUDA 9.0
cuDNN 7.1
GPU GeForce GTX 1070

**Comment by gunan** [comment]
Since this is under contrib, and looking at the history, I am marking this contributions welcome as it looks like we wont be able to get to this.

**Comment by dynamicwebpaige** [comment]
Closing out this issue, as `tf.contrib.image` has been deprecated as part of TensorFlow 2.0. If you run into similar snags with `tf.image.*` ops when using Windows with CUDA enabled, please feel free to open a new ticket. Thanks! 🙂
